webapp/tests/model/test_eligibility_data.py

```diff
@@ -1176,33 +1182,60 @@ def test_if_other_income_valid_and_foreign_country_no_and_registration_method_se
             with patch('app.model.eligibility_data.OtherIncomeEligibilityData.__init__',
                        MagicMock(return_value=None)):
                 ForeignCountrySuccessEligibility.parse_obj(valid_data)
-        except ValidationError as e:
+        except ValidationError:
+            self.fail("ForeignCountryEligibility.parse_obj should not raise validation error")
+
+    def test_if_other_income_valid_and_foreign_country_no_but_abrufcode_none_then_raise_validation_error(self):
+        non_valid_data = {'foreign_country_eligibility': 'no', 'elster_abrufcode_eligibility': 'none'}
+        with patch('app.model.eligibility_data.OtherIncomeEligibilityData.__init__', MagicMock(return_value=None)):
+            self.assertRaises(ValidationError, ForeignCountrySuccessEligibility.parse_obj, non_valid_data)
+
+    def test_if_other_income_valid_and_foreign_country_no_and_abrufcode_set_to_not_none_then_raise_no_validation_error(self):
+        valid_data = {'foreign_country_eligibility': 'no', 'elster_abrufcode_eligibility': 'NOT_NONE'}
+        try:
+            with patch('app.model.eligibility_data.OtherIncomeEligibilityData.__init__',
+                       MagicMock(return_value=None)):
+                ForeignCountrySuccessEligibility.parse_obj(valid_data)
+        except ValidationError:
```

Comment:
I might have a problem understanding why this should not raise a validation error? In my head it should only raise no validation error if `elster_abrufcode_eligibility = 'no'`?

Reply:
I think we should talk about it before the discussion gets too big in here